src/Agent.Worker/Build/BuildDirectoryManager.cs

```diff
@@ -122,7 +122,25 @@ public sealed class BuildDirectoryManager : AgentService, IBuildDirectoryManager
             // Set the default clone path for each repository (the Checkout task may override this later)
             foreach (var repository in repositories)
             {
-                var repoPath = GetDefaultRepositoryPath(executionContext, repository, newConfig.SourcesDirectory);
+                String repoPath;
+                if (RepositoryUtil.HasMultipleCheckouts(executionContext.JobSettings))
+                {
+                    // If we have multiple checkouts they should all be rooted to the sources directory (_work/1/s/repo1)
+                    var repoSourceDirectory = newConfig?.RepositoryTrackingInfo.Where(item => string.Equals(item.Identifier, repository.Alias, StringComparison.OrdinalIgnoreCase)).Select(item => item.SourcesDirectory).FirstOrDefault();
+                    if (repoSourceDirectory != null)
+                    {
+                        repoPath = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Work), repoSourceDirectory);
```

Comment:
Nit: could we avoid duplication here?

Reply:
Am I understand you correctly that you mean duplication of _HostContext.GetDirectory(WellKnownDirectory.Work)_ part?